extensions/map-layers/src/mapLayers.ts

```diff
@@ -2,33 +2,39 @@
 * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
-import { IModelApp } from "@itwin/core-frontend";
+import { IModelApp, UserPreferencesAccess } from "@itwin/core-frontend";
 import { MapLayersUiItemsProvider, MapLayersWidgetControl } from "./ui/MapLayersUiItemsProvider";
 import { UiItemsManager } from "@itwin/appui-abstract";
 import { ConfigurableUiManager } from "@itwin/appui-react";
 
-/**
- * MapLayersApi is use when the package is used as a dependency to another app and not used as an extension.
+/** MapLayersUI is use when the package is used as a dependency to another app.
  * '''ts
- *  // if registerItemsProvider is false the MapLayersWidgetControl control will be registered with appui-react's ConfigurableUiManager
- *  // so it can be explicitly added to a stage via a FrontstageDef.
- *  await MapLayersUI.initialize (registerItemsProvider);
+ *  await MapLayersUI.initialize(registerItemsProvider);
  * '''
  * @beta
  */
 export class MapLayersUI {
   private static _defaultNs = "mapLayers";
   private static _uiItemsProvider: MapLayersUiItemsProvider;
 
-  /** Used to initialize the MapLayersAPI when used as a package. If `registerItemsProvider` is true then the
-   * UiItemsProvider will automatically insert the UI items into the host applications UI. If it is false then
-   * explicitly add widget definition to a specific FrontStage definition using the following syntax.
-   * ``` tsx
-   * <Widget id={MapLayersWidgetControl.id} label={MapLayersWidgetControl.label} control={MapLayersWidgetControl}
+  private static _iTwinConfig?: UserPreferencesAccess;
```

Comment:
So by defining the config at the widget level, are you expecting consumers to provide an implementation that might be different than IModelApp.userPreferences ?  If not then shouldn't we just call directly IModelApp.userPreferences directly inside the widget?

Reply:
I would expect a different implementation since the user preferences are only stored at a per-user level rather than today all of the map layers information is stored and shared on a per-application level. It more similarly follows with the Workspaces that we're working on right now and I think we'd eventually want to switch to using that.